blockdevice/blockdevice.go

```diff
@@ -469,3 +469,5 @@ const (
 // of its parent device. We will get /dev/sda is the parent of /dev/sda1.
 // This will be used to query the cache and get/generate the UUID of /dev/sda.
 type Hierarchy map[string]BlockDevice
+
+var AllBlockDevices BlockDevice
```

Comment:
Why is a variable created here? Cant the struct be directly used?

Reply:
using a variable since we'll need to refer to its address in other parts of the code. I'm not sure what you mean by direct struct.

Reply:
Then we should declare the  variable at some other part of the code. The `blockdevice.go` file is meant for only definitions.

Reply:
I see. But I feel it would be better to have it in the blockdevice package so that all other parts of the code can use the same variable uniformly. If it will be used by probe package alone, then yes, it will be better to declare the variable in the probe package

Reply:
Nope. All parts of the code having equal access to modify is risky. Since we are accessing it from multiple go routines we need to make sure the modifications happen only in the intended way. You can see the same with BDHierarchy, if it was present along with the struct definitions it would be easy to access but maintaining the concurrent access becomes difficult. So declare it closest to where it is being used.

Reply:
> Nope. All parts of the code having equal access to modify is risky. Since we are accessing it from multiple go routines we need to make sure the modifications happen only in the intended way.

Oh, so this variable is more of a placeholder. It is meant to be read-only since we only care about the address of the variable. The content stored inside doesn't matter and that why I have simply declared it. 

> So declare it closest to where it is being used.

All right. Though I feel that since its read-only, it will be safe from concurrent patterns.